registry-library/library/library.go

```diff
@@ -103,6 +103,10 @@ type RegistryFilter struct {
 	// only major version and minor version are required. e.g. 2.1, 2.2 ect. service version should not be provided.
 	// will only be applied if `NewIndexSchema=true`
 	MaxSchemaVersion string
+	// Deprecated is set to filter devfile index for stacks having the "Deprecated" tag inside their default set of tags
+	// or not. the only acceptable values are "true"/"false". in case the value is "true" it will only include only
+	// deprecated stacks, if is "false" only non deprecated. will only be applied if `NewIndexSchema=true`
+	Deprecated string
```

Comment:
@thepetk I wondering if it would be good to use an "enum" for `RegistryFilter.Deprecated`, i.e. we create a `DeprecatedFilter` type that holds `string` values which `RegistryFilter.Deprecated` is typed as and create two constants:

```go
const DeprecatedFilterTrue DeprecatedFilter = "true"
const DeprecatedFilterFalse DeprecatedFilter = "false"
```

WDYT?

Reply:
I think makes sense! I've updated the code to reflect the suggested updates